shared_utils.py

```diff
@@ -16,30 +25,36 @@
 logging.basicConfig(level=LOG_LEVEL, format=LOG_FORMAT)
 
 
+def log_and_exit(message: str):
+    """
+    Log an error message and then exit with EXIT_FAILURE.
+    """
+    logging.error(message)
+    raise SystemExit(1)
+
```

Comment:
@Timmmm could you remove this as this is already been taken care as a part of #309 to fix `log_and_exit` things.
Thanks!

Reply:
I will if that gets merged first, though maybe it would be better handled in #293 because I updated the file to use it everywhere in that PR. And it was first :-P

Reply:
Oh! I see, Thanks for letting me know. I was not not aware though.
Well in that case, I will resolve the conflicts/ update the PR with master branch for #309 , if in any case, as such arises.